### ISSUE TITLE
Unable to launch connectomemapper GUI

### ISSUE THREAD

**Issue by amitjc** [connectomicslab/connectomemapper3#29]
Hello

I'll be highly obliged if someone help me reg. launching GUI

please find the log below:

(py27cmp-gui) amitjc@AMiT:/usr/local/connectomemapper/connectomemapper3$ cmpbidsappmanager
Traceback (most recent call last):
  File "/usr/local/bin/cmpbidsappmanager", line 15, in <module>
    from bids import BIDSLayout
ImportError: No module named bids

Please Advice.

Thanks and Regards,
Amit.

**Comment by sebastientourbier** [comment]
Hi Amit,

Just to have a better idea of what can be wrong, please could you give me the version tag of connectomemapper3 you are using? Also, what is the result of `conda list` right after you activate the `py27cmp-gui` conda environment?

**Comment by sebastientourbier** [comment]
Any update?

**Comment by amitjc** [comment]
Sorry this got delayed.


(py27cmp-gui) amitjc@AMiT:~$ conda list
# packages in environment at /home/amitjc/miniconda2/envs/py27cmp-gui:
#
# Name                    Version                   Build  Channel
backports                 1.0                        py_2    anaconda
backports.functools_lru_cache 1.6.1                      py_0    anaconda
backports_abc             0.5                      py27_0    anaconda
blas                      1.0                         mkl    anaconda
bz2file                   0.98                     py27_1    anaconda
ca-certificates           2020.1.1                      0    anaconda
cairo                     1.14.12              h8948797_3    anaconda
certifi                   2019.11.28               py27_0    anaconda
clangdev                  8.0.0                hc9558a2_2    conda-forge
click                     7.0                      py27_0    anaconda
configparser              4.0.2                    py27_0    anaconda
cycler                    0.10.0                   py27_0    anaconda
dbus                      1.13.12              h746ee38_0    anaconda
decorator                 4.4.2                      py_0    anaconda
expat                     2.2.6                he6710b0_0    anaconda
fontconfig                2.13.0               h9420a91_0    anaconda
freetype                  2.9.1                h8a8886c_1    anaconda
fribidi                   1.0.5                h7b6447c_0    anaconda
funcsigs                  1.0.2                    py27_0    anaconda
functools32               3.2.3.2                  py27_1    anaconda
future                    0.18.2                   py27_0    anaconda
futures                   3.3.0                    py27_0    anaconda
git                       2.23.0          pl526hacde149_0    anaconda
git-annex                 7.20190219          h4a95a7c_12    conda-forge
glib                      2.56.2               hd408876_0    anaconda
gmp                       6.1.2                hb3b607b_0    anaconda
gnutls                    3.6.5             hd3a4fd2_1002    conda-forge
graphite2                 1.3.13               h23475e2_0    anaconda
graphviz                  2.40.1               h21bd128_2    anaconda
gst-plugins-base          1.14.0               hbbd80ab_1    anaconda
gstreamer                 1.14.0               hb453b48_1    anaconda
h5py                      2.9.0            py27h7918eee_0    anaconda
harfbuzz                  1.8.8                hffaf4a1_0    anaconda
hdf5                      1.10.4               hb1b8bf9_0    anaconda
html5lib                  1.0.1                    py27_0    anaconda
icu                       58.2                 h211956c_0    anaconda
intel-openmp              2020.0                      166    anaconda
isodate                   0.6.0                      py_1    anaconda
jpeg                      9b                   habf39ab_1    anaconda
keepalive                 0.5                        py_1    conda-forge
kiwisolver                1.1.0            py27he6710b0_0    anaconda
krb5                      1.17.1               h173b8e3_0    anaconda
libcurl                   7.68.0               h20c2e04_0    anaconda
libedit                   3.1.20181209         hc058e9b_0    anaconda
libffi                    3.2.1                h4deb6c0_3    anaconda
libgcc-ng                 9.1.0                hdf63c60_0    anaconda
libgfortran-ng            7.3.0                hdf63c60_0    anaconda
libiconv                  1.15                 h63c8f33_5    anaconda
libpng                    1.6.37               hbc83047_0    anaconda
libssh2                   1.9.0                h1ba5d50_1    anaconda
libstdcxx-ng              9.1.0                hdf63c60_0    anaconda
libtiff                   4.1.0                h2733197_0    anaconda
libuuid                   1.0.3                h1bed415_2    anaconda
libxcb                    1.13                 h1bed415_1    anaconda
libxml2                   2.9.9                hea5a465_1    anaconda
libxslt                   1.1.33               h7d1a2b0_0    anaconda
linecache2                1.0.0                    py27_0    anaconda
llvmdev                   8.0.0                hc6ec683_0    anaconda
lxml                      4.4.2            py27hefd8a0e_0    anaconda
matplotlib                2.2.3            py27hb69df0a_0    anaconda
mkl                       2019.5                      281    anaconda
mkl-service               2.3.0            py27he904b0f_0    anaconda
mkl_fft                   1.0.15           py27ha843d7b_0    anaconda
mkl_random                1.1.0            py27hd6b4f25_0    anaconda
ncurses                   6.2                  he6710b0_0    anaconda
nettle                    3.4.1             h1bed415_1002    conda-forge
networkx                  2.2                      py27_1    anaconda
neurdflib                 5.0.1                      py_0    conda-forge
nibabel                   2.5.1                      py_0    conda-forge
nipype                    1.1.8                    py27_0    conda-forge
numpy                     1.15.4           py27h7e9f1db_0    anaconda
numpy-base                1.15.4           py27hde5b4d6_0    anaconda
openssl                   1.1.1                h7b6447c_0    anaconda
packaging                 20.3                       py_0    anaconda
pango                     1.42.4               h049681c_0    anaconda
pcre                      8.43                 he6710b0_0    anaconda
perl                      5.26.2               h14c3975_0    anaconda
pip                       19.0.3                   py27_0    anaconda
pixman                    0.38.0               h7b6447c_0    anaconda
popt                      1.16              h299ea2f_2002    conda-forge
prov                      1.5.3                      py_1    conda-forge
psutil                    5.6.7            py27h7b6447c_0    anaconda
pydicom                   1.4.2                      py_0    conda-forge
pydot                     1.4.1                    py27_0    anaconda
pydotplus                 2.0.2                    py27_1    anaconda
pyface                    6.1.2                    py27_0    anaconda
pygments                  2.5.2                      py_0    anaconda
pyparsing                 2.4.6                      py_0    anaconda
pyqt                      5.9.2            py27h22d08a2_1    anaconda
pyside2                   5.9.0a1          py27h4dc837a_0    conda-forge
python                    2.7.15               h9bab390_6    anaconda
python-dateutil           2.8.1                      py_0    anaconda
pytz                      2019.3                     py_0    anaconda
qt                        5.9.7                h5867ecd_1    anaconda
rdflib                    4.2.2                 py27_1000    conda-forge
readline                  7.0                  h7b6447c_5    anaconda
rsync                     3.1.3             hed695b0_1002    conda-forge
scipy                     1.2.1            py27h7c811a0_0    anaconda
setuptools                44.0.0                   py27_0    anaconda
simplejson                3.17.0           py27h7b6447c_0    anaconda
singledispatch            3.4.0.3                  py27_0    anaconda
sip                       4.19.13          py27he6710b0_0    anaconda
six                       1.13.0                   py27_0    anaconda
sparqlwrapper             1.8.2                 py27_1000    conda-forge
sqlite                    3.31.1               h7b6447c_0    anaconda
subprocess32              3.5.4            py27h7b6447c_0    anaconda
tk                        8.6.8                hbc83047_0    anaconda
tornado                   5.1.1            py27h7b6447c_0    anaconda
traceback2                1.4.0                    py27_0    anaconda
traits                    5.1.2            py27h7b6447c_0    anaconda
traitsui                  6.0.0                    py27_1    anaconda
unittest2                 1.1.0                    py27_0    anaconda
webencodings              0.5.1                    py27_1    anaconda
wheel                     0.33.6                   py27_0    anaconda
xvfbwrapper               0.2.9                 py27_1001    conda-forge
xz                        5.2.4                h14c3975_4    anaconda
zlib                      1.2.11               h7b6447c_3    anaconda
zstd                      1.3.7                h0b5b093_0    anaconda

**Comment by sebastientourbier** [comment]
Ok seems pybids is indeed not installed. This is strange as it appears in the environment.yml file.

Could you try? 

```
(py27cmp-gui) amitjc@AMiT:~$ pip install pybids==0.9.1
```

Let me know if it works.

Also, just to help me understand what could have been gone wrong, which OS are using? Also which version of CMP3 are you using? 
I just released a new version yesterday with multiple bug fixes and improved documentation.

**Comment by amitjc** [comment]
Trying the solution.

Further details:

Os: Ubuntu 19.10

IS there a command to look for CMP 3 version?

**Comment by amitjc** [comment]
Hello Sébastien,

Couldn't launch the GUI post pybids installation, with similar error.

Please find the log below.

Thank you.

Best Regards,
Amit.

(py27cmp-gui) amitjc@AMiT:~$ pip install pybids==0.9.1
DEPRECATION: Python 2.7 will reach the end of its life on January 1st,
2020. Please upgrade your Python as Python 2.7 won't be maintained after
that date. A future version of pip will drop support for Python 2.7.
Collecting pybids==0.9.1
  Downloading
https://files.pythonhosted.org/packages/91/30/5e92a443d913cadd0b42125b71c62d6b798a1a0c45e9994eab0a8ba2adf8/pybids-0.9.1.tar.gz
(1.5MB)
    100% |████████████████████████████████| 1.5MB 90kB/s
Collecting num2words (from pybids==0.9.1)
  Downloading
https://files.pythonhosted.org/packages/33/db/76f1151a1b0cfad532d41021b77cd231495bf72c47618166f92dcdff2ebe/num2words-0.5.10.tar.gz
(114kB)
    100% |████████████████████████████████| 122kB 146kB/s
Requirement already satisfied: numpy in
./.local/lib/python2.7/site-packages (from pybids==0.9.1) (1.16.6)
Requirement already satisfied: scipy in
./.local/lib/python2.7/site-packages (from pybids==0.9.1) (1.2.3)
Requirement already satisfied: pandas>=0.23.0 in
./.local/lib/python2.7/site-packages (from pybids==0.9.1) (0.24.2)
Requirement already satisfied: nibabel>=2.1 in
./.local/lib/python2.7/site-packages (from pybids==0.9.1) (2.5.1)
Collecting patsy (from pybids==0.9.1)
  Downloading
https://files.pythonhosted.org/packages/ea/0c/5f61f1a3d4385d6bf83b83ea495068857ff8dfb89e74824c6e9eb63286d8/patsy-0.5.1-py2.py3-none-any.whl
(231kB)
    100% |████████████████████████████████| 235kB 142kB/s
Collecting bids-validator (from pybids==0.9.1)
  Downloading
https://files.pythonhosted.org/packages/81/7f/dc565a737d36e6220760f5a5bfaff580441ec18f2ec073a296b5a5b655d4/bids_validator-1.4.4-py2.py3-none-any.whl
Collecting SQLAlchemy (from pybids==0.9.1)
  Downloading
https://files.pythonhosted.org/packages/8c/30/4134e726dd5ed13728ff814fa91fc01c447ad8700504653fe99d91fdd34b/SQLAlchemy-1.3.15.tar.gz
(6.1MB)
    100% |████████████████████████████████| 6.1MB 71kB/s
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
    Preparing wheel metadata ... done
Collecting docopt>=0.6.2 (from num2words->pybids==0.9.1)
  Downloading
https://files.pythonhosted.org/packages/a2/55/8f8cab2afd404cf578136ef2cc5dfb50baa1761b68c9da1fb1e4eed343c9/docopt-0.6.2.tar.gz
Requirement already satisfied: pytz>=2011k in
./.local/lib/python2.7/site-packages (from pandas>=0.23.0->pybids==0.9.1)
(2019.3)
Requirement already satisfied: python-dateutil>=2.5.0 in
./miniconda2/envs/py27cmp-gui/lib/python2.7/site-packages (from
pandas>=0.23.0->pybids==0.9.1) (2.8.1)
Requirement already satisfied: bz2file; python_version < "3.0" in
./.local/lib/python2.7/site-packages (from nibabel>=2.1->pybids==0.9.1)
(0.98)
Requirement already satisfied: six>=1.3 in
./miniconda2/envs/py27cmp-gui/lib/python2.7/site-packages (from
nibabel>=2.1->pybids==0.9.1) (1.13.0)
Building wheels for collected packages: SQLAlchemy
  Building wheel for SQLAlchemy (PEP 517) ... done
  Stored in directory:
/home/amitjc/.cache/pip/wheels/24/7f/c8/8d776916c91c613d593f39a98cad373442c7f77027340eb1ec
Successfully built SQLAlchemy
Building wheels for collected packages: pybids, num2words, docopt
  Building wheel for pybids (setup.py) ... done
  Stored in directory:
/home/amitjc/.cache/pip/wheels/a3/a3/49/c66a2dfc5e594e9177890d1da462cafa0698a512e2d628e293
  Building wheel for num2words (setup.py) ... done
  Stored in directory:
/home/amitjc/.cache/pip/wheels/2c/6c/20/c2cbd7eb4f66871921b5535090e160375621d571375cd4d1bd
  Building wheel for docopt (setup.py) ... done
  Stored in directory:
/home/amitjc/.cache/pip/wheels/9b/04/dd/7daf4150b6d9b12949298737de9431a324d4b797ffd63f526e
Successfully built pybids num2words docopt
Installing collected packages: docopt, num2words, patsy, bids-validator,
SQLAlchemy, pybids
Successfully installed SQLAlchemy-1.3.15 bids-validator-1.4.4 docopt-0.6.2
num2words-0.5.10 patsy-0.5.1 pybids-0.9.1
(py27cmp-gui) amitjc@AMiT:~$ cmpbidsappmanager
Traceback (most recent call last):
  File "/usr/local/bin/cmpbidsappmanager", line 15, in <module>
    from bids import BIDSLayout
ImportError: No module named bids

**Comment by sebastientourbier** [comment]
Hi Amit,

By looking at the error:
```
Traceback (most recent call last):
  File "/usr/local/bin/cmpbidsappmanager", line 15, in <module>
    from bids import BIDSLayout
ImportError: No module named bids
```
It seems the cmpbidsappmanager command is not the one installed in your `py27cmp-gui` conda environment but rather the one located at `/usr/local/bin/cmpbidsappmanager`.
I suspect you might have performed the installation of the GUI (command `python setup_gui.py install`) outside the conda environment.
Could you delete `cmpbidsappmanager` from /usr/local/bin and try to install the GUI while being in the conda environment, i.e.
```
(py27cmp-gui) amitjc@AMiT:~$ cd /path/to/connectomemapper3
(py27cmp-gui) amitjc@AMiT:~/path/to/connectomemapper3$ python setup_gui.py install
```
To get the version of CMP3 you have cloned from github, you can run `python /path/to/connectomemapper3/get_version.py`

**Comment by amitjc** [comment]
Hello Sébastien,

Couldn't get CMP3 gui to run with above.

Please find the log below (version can be found in the log) and advice.

Thanks.

Best Regards,
Amit.




(base) amitjc@AMiT:/usr/local/connectomemapper/connectomemapper3$ conda
activate py27cmp-gui
(py27cmp-gui) amitjc@AMiT:/usr/local/connectomemapper/connectomemapper3$
python setup_gui.py install
running install
running build
running build_py
creating build/lib
error: could not create 'build/lib': Permission denied
(py27cmp-gui) amitjc@AMiT:/usr/local/connectomemapper/connectomemapper3$
sudo python setup_gui.py install
[sudo] password for amitjc:
running install
running build
running build_py
running build_scripts
running install_lib
running install_scripts
copying build/scripts-2.7/cmpbidsappmanager -> /usr/local/bin
changing mode of /usr/local/bin/cmpbidsappmanager to 755
changing mode of /usr/local/bin/showmatrix_gpickle to 755
running install_egg_info
Removing
/usr/local/lib/python2.7/dist-packages/cmpbidsappmanager-v3.0.0_beta_20200227.egg-info
Writing
/usr/local/lib/python2.7/dist-packages/cmpbidsappmanager-v3.0.0_beta_20200227.egg-info
(py27cmp-gui) amitjc@AMiT:/usr/local/connectomemapper/connectomemapper3$
get_version.py
get_version.py: command not found
(py27cmp-gui) amitjc@AMiT:/usr/local/connectomemapper/connectomemapper3$
usr/local/connectomemapper/connectomemapper3/get_version.py
bash: usr/local/connectomemapper/connectomemapper3/get_version.py: No such
file or directory
(py27cmp-gui) amitjc@AMiT:/usr/local/connectomemapper/connectomemapper3$
cmpbidsappmanager
Traceback (most recent call last):
  File "/usr/local/bin/cmpbidsappmanager", line 15, in <module>
    from bids import BIDSLayout
ImportError: No module named bids
(py27cmp-gui) amitjc@AMiT:/usr/local/connectomemapper/connectomemapper3$

**Comment by sebastientourbier** [comment]
Hi Amit,

It seems because of some previous wrong manipulations, some folders and installations are messed up.

When you did:
```
(py27cmp-gui) amitjc@AMiT:/usr/local/connectomemapper/connectomemapper3$
sudo python setup_gui.py install
[sudo] password for amitjc:
running install
running build
running build_py
running build_scripts
running install_lib
running install_scripts
copying build/scripts-2.7/cmpbidsappmanager -> /usr/local/bin
```
you managed to solve the issue for the permission denied error but in fact, by running the installation in a privileged mode (sudo), it does not install the GUI in the conda environment, but in /usr/local/bin.

In case, you can know the exact location of the created `py27cmp-gui` conda environment by running the command `conda env list`.

What I would suggest is to restart the clone of the repository and the installation of the GUI from the beginning but without any need to reinstall miniconda2 and the `py27cmp-gui` environment.
This consists in the following steps:

1. Completely remove the cloned repository of connectomemapper3:
```
$ (sudo) rm -R /usr/local/connectomemapper/connectomemapper3
```
Note: you might need to run the `rm` in privileged mode if you get again the permission denied error

2. Clone again the github repo, but this time somewhere in your user home which will prevent some permission related error (for instance in ~/Softwares):
```
$ mkdir -p ~/Softwares
$ cd ~/Softwares
$ git clone https://github.com/connectomicslab/connectomemapper3.git connectomemapper3

3. Create a branch and checkout the code corresponding to the latest beta candidate release:
```
$ cd ~/Softwares/connectomemapper3
$ git checkout tags/v3.0.0-beta-RC1 -b v3.0.0-beta-RC1
```

4. Activate your already installed conda environment:
```
$ conda activate py27cmp-gui
```

5. Install the Connectome Mapper BIDS App Manager :
```
(py27cmp-gui)$ cd ~/Softwares/connectomemapper3
(py27cmp-gui)$ python setup_gui.py install
```

Then, it should work.

**Comment by amitjc** [comment]
Hello Sébastien,

Appreciate your constant support. Many thanks.

Tried the above.

Running into errors. 

Please find the log below.

Thanks Again.

Best Regards,
Amit.


(py27cmp-gui) amitjc@AMiT:~/Softwares/connectomemapper3$ cmpbidsappmanager
Graphical Backend : qt4
200404-00:13:59,912 nipype.interface INFO:
	 We advise you to upgrade DIPY version. This upgrade will activate RecoBundles, LabelsBundles, DeterministicTracking.
200404-00:13:59,916 nipype.interface INFO:
	 We advise you to upgrade DIPY version. This upgrade will activate DKIModel, MapmriModel, DTIModel, CSAModel, CSDModel.
Traceback (most recent call last):
  File "/home/amitjc/miniconda2/envs/py27cmp-gui/bin/cmpbidsappmanager", line 21, in <module>
    from cmp.bidsappmanager import gui
  File "/home/amitjc/miniconda2/envs/py27cmp-gui/lib/python2.7/site-packages/cmp/bidsappmanager/gui.py", line 40, in <module>
    import cmp.bidsappmanager.project as project
  File "/home/amitjc/miniconda2/envs/py27cmp-gui/lib/python2.7/site-packages/cmp/bidsappmanager/project.py", line 36, in <module>
    from cmp.bidsappmanager.pipelines.functional import fMRI as FMRI_pipeline
  File "/home/amitjc/miniconda2/envs/py27cmp-gui/lib/python2.7/site-packages/cmp/bidsappmanager/pipelines/functional/fMRI.py", line 29, in <module>
    from cmp.bidsappmanager.pipelines.anatomical.anatomical import AnatomicalPipelineUI
  File "/home/amitjc/miniconda2/envs/py27cmp-gui/lib/python2.7/site-packages/cmp/bidsappmanager/pipelines/anatomical/anatomical.py", line 32, in <module>
    from cmp.pipelines.common import Pipeline
  File "/home/amitjc/miniconda2/envs/py27cmp-gui/lib/python2.7/site-packages/cmp/pipelines/common.py", line 34, in <module>
    import apptools.io.api as io
ImportError: No module named apptools.io.api
(py27cmp-gui) amitjc@AMiT:~/Softwares/connectomemapper3$

**Comment by amitjc** [comment]
Hello Sébastien,

Appreciate your constant support. Many thanks.

Tried the above.

Running into errors.

Please find the log below.

Thanks Again.

Best Regards,
Amit.



(py27cmp-gui) amitjc@AMiT:~/Softwares/connectomemapper3$ cmpbidsappmanager
Graphical Backend : qt4
200404-00:13:59,912 nipype.interface INFO:
We advise you to upgrade DIPY version. This upgrade will activate
RecoBundles, LabelsBundles, DeterministicTracking.
200404-00:13:59,916 nipype.interface INFO:
We advise you to upgrade DIPY version. This upgrade will activate DKIModel,
MapmriModel, DTIModel, CSAModel, CSDModel.
Traceback (most recent call last):
  File "/home/amitjc/miniconda2/envs/py27cmp-gui/bin/cmpbidsappmanager",
line 21, in <module>
    from cmp.bidsappmanager import gui
  File
"/home/amitjc/miniconda2/envs/py27cmp-gui/lib/python2.7/site-packages/cmp/bidsappmanager/gui.py",
line 40, in <module>
    import cmp.bidsappmanager.project as project
  File
"/home/amitjc/miniconda2/envs/py27cmp-gui/lib/python2.7/site-packages/cmp/bidsappmanager/project.py",
line 36, in <module>
    from cmp.bidsappmanager.pipelines.functional import fMRI as
FMRI_pipeline
  File
"/home/amitjc/miniconda2/envs/py27cmp-gui/lib/python2.7/site-packages/cmp/bidsappmanager/pipelines/functional/fMRI.py",
line 29, in <module>
    from cmp.bidsappmanager.pipelines.anatomical.anatomical import
AnatomicalPipelineUI
  File
"/home/amitjc/miniconda2/envs/py27cmp-gui/lib/python2.7/site-packages/cmp/bidsappmanager/pipelines/anatomical/anatomical.py",
line 32, in <module>
    from cmp.pipelines.common import Pipeline
  File
"/home/amitjc/miniconda2/envs/py27cmp-gui/lib/python2.7/site-packages/cmp/pipelines/common.py",
line 34, in <module>
    import apptools.io.api as io
ImportError: No module named apptools.io.api
(py27cmp-gui) amitjc@AMiT:~/Softwares/connectomemapper3$

**Comment by sebastientourbier** [comment]
I just successfully tested a fresh new installation of the up-coming version of CMP3 (#33, commit 5a2e00cc5903079c970b4cfe89f8458962213a58) along with the installation of new conda environment using the new environment.yml file. Thus, I consider this issue as solved and will be closed by #33 when merged.
If you encounter the same issue with the new version, please reopen this issue

**Comment by amitjc** [comment]
Thank you for the update.

Shall give this a try afresh.

Best Regards,
Amit.
